### PR TITLE
Replatforming ADR

### PR DESCRIPTION
**Pull request by danworth**
Add an ADR stating the intention to move GOV.UK Forms to AWS ECS Fargate.

# PR Checklist

- [x] If you are proposing a new decision record document, used the right template for that
      - ([ADR](https://github.com/alphagov/forms/blob/main/ADR/ADRXXX-architecture-decision-record-template.md), [decision-record](https://github.com/alphagov/forms/blob/main/decision-record/DRXXX-decision-record-template.md), engagement, [research](https://github.com/alphagov/forms/blob/main/research/YYYY-MM-DD-template.md))
- [x] Set yourself as the Assignee
- [x] Tag anyone you would like to review, or @forms-design or @forms-devs
- [ ] Fill in the template below

### CODE REVIEW

File: ADR/ADR009-replatforming-to-ECS.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+Date: 2022-10-27
+
+## Status
+
+Accepted
+
+## Context
+
+GOV.UK Forms currently runs on GOV.UK PaaS which is being decommissioned in
```
Comment (james-sheppard-gov-uk) [inline ADR/ADR009-replatforming-to-ECS.md:11]: How about a link to the "Why we’ve decided to decommission GOV.UK PaaS" blog post from [GOV.UK PaaS which is being decommissioned in December 2023](https://gds.blog.gov.uk/2022/07/12/why-weve-decided-to-decommission-gov-uk-paas-platform-as-a-service/)

File: ADR/ADR009-replatforming-to-ECS.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+of other GDS products. We can make use of existing experience and expertise.
+- The current and expected future infrastructure requirements of GOV.UK Forms can be met.
+- Fargate deployment mode takes care of the underlying VM management to reduce
+the maintenance burden.
+- High level of confidence that GOV.UK Forms will work as-is with AWS services because they
+are already being used by the GOV.UK PaaS implementation (e.g. RDS and
+Elasticache).
+- Feature development will continue throughout the migration with GOV.UK Forms
+being dual deployed to PaaS and ECS until cut-over.
+- The GOV.UK Forms team will need to manage more infrastructure since GOV.UK PaaS
```
Comment (james-sheppard-gov-uk) [inline ADR/ADR009-replatforming-to-ECS.md:33]: I gather the PaaS team will be providing some guidance on what this is shortly...

File: ADR/ADR009-replatforming-to-ECS.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+## Decision
+
+Move GOV.UK Forms onto AWS Elastic Container Service (ECS) with Fargate and supporting
+AWS services such as RDS and Elasticache. All infrastructure to be managed with
+Terraform. Continuous integration and deployment patterns will be implemented
+(technology choices are TBD).
+
+## Consequences
+
```
Comment (james-sheppard-gov-uk) [inline ADR/ADR009-replatforming-to-ECS.md:22]: - The effort required to design and migrate to new infrastructure will reduce the amount of roadmap features that can be developed during the migration.

File: ADR/ADR009-replatforming-to-ECS.md (hunk 2 above)
Comment (danworth) [reply ADR/ADR009-replatforming-to-ECS.md:33]: 👍 essentially we'll manage all of the infrastructure that Forms needs to run. Most of the infra that PaaS currently manages for us has a direct replacement as an AWS service we can manage ourselves (in most cases it is the same underlying AWS service, e.g. RDS and elasticache).

**Review by james-sheppard-gov-uk — APPROVED** [review]

**Pull request merged**